src/hls-video-element.js

```diff
@@ -0,0 +1,99 @@
+import { SuperVideoElement } from 'super-media-element';
+import Hls from 'hls.js';
+
+class HlsVideoElement extends SuperVideoElement {
+  async attributeChangedCallback(attrName, oldValue, newValue) {
+    switch (attrName) {
+      case 'src': {
+        this.load();
+        return;
+      }
+    }
+
+    super.attributeChangedCallback(attrName, oldValue, newValue);
+  }
+
+  disconnectedCallback() {
+    this.#destroy();
+  }
+
+  #destroy() {
+    if (this.api) {
+      this.api.detachMedia();
+      this.api.destroy();
+      this.api = null;
+    }
+  }
+
+  get src() {
+    return this.getAttribute('src');
+  }
+
+  set src(val) {
+    if (val == this.src) return;
+    this.setAttribute('src', val);
+  }
+
+  async load() {
+    this.#destroy();
+
+    // Wait 1 tick to allow other attributes to be set.
+    await Promise.resolve();
+
+    if (!this.src) {
+      return;
+    }
+
+    if (Hls.isSupported()) {
+      this.api = new Hls();
```

Comment:
I like exposing the Hls.js API like this. 👍 Ideally we can expose more things via media-ui-extensions over time so you don't need it so much. I kind of like `api` as the name, but also this isn't the only API. Is the hope to standardize this property between custom media elements?

Reply:
yes, I've been using it for playerx to access the internal API's of the different types of video API's.
a bit like nativeEl but for Vimeo, YT, Wistia this accessed their custom API's.